flatland_server/include/flatland_server/profiler.h

```diff
@@ -0,0 +1,99 @@
+#ifndef FLATLAND_PROFILER_H
+#define FLATLAND_PROFILER_H
+
+#define PROFILER_ON false
+#define PROFILER_OUTPUT_PATH "/tmp/flatland_profile_output.log"
+
+#if PROFILER_ON == true
+#define START_PROFILE(timekeeper, name) timekeeper.profiler_.get(name).start()
+#define END_PROFILE(timekeeper, name) timekeeper.profiler_.get(name).end()
+#define PRINT_ALL_PROFILES(timekeeper) timekeeper.profiler_.print()
+#else
+#define START_PROFILE(timekeeper, name)
+#define END_PROFILE(timekeeper, name)
+#define PRINT_ALL_PROFILES(timekeeper)
+#endif
+
+#include <cassert>
+#include <chrono>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <unordered_map>
+
+namespace flatland_server {
+
+class Profile {
+ public:
+  Profile()
+      : lapse_count_(0),
+        started_(false),
+        start_time_(std::chrono::high_resolution_clock::now()),
+        total_duration_(std::chrono::high_resolution_clock::duration::zero()) {}
+
+  void start() {
+    started_ = true;
+    start_time_ = std::chrono::high_resolution_clock::now();
+  }
+
+  void end() {
+    assert(started_);
+    lapse_count_++;
+    total_duration_ += std::chrono::duration_cast<std::chrono::microseconds>(
+        std::chrono::high_resolution_clock::now() - start_time_);
+  }
+
+  int getLapseCount() const { return lapse_count_; }
+
+  long getTotalDuration() const {
+    return std::chrono::duration_cast<std::chrono::microseconds>(
+               total_duration_)
+        .count();
+  }
+
+ private:
+  int lapse_count_;
+  bool started_;
+  std::chrono::high_resolution_clock::time_point start_time_;
+  std::chrono::high_resolution_clock::duration total_duration_;
+};
+
+class Profiler {
+ public:
+  Profile& get(const std::string& profile_name) {
```

Comment:
I'm not a big fan of using strings...for anything. For now I think it is ok, if we want to measure more accurately we should replace them though.